test/tap/repo.js

```diff
@@ -132,6 +132,79 @@ test('npm repo test-repo-url-ssh - non-github (ssh://)', function (t) {
   })
 })
 
+/* ----- Test by new mock registry: BEGIN ----- */
+
+const Tacks = require('tacks')
+const mockTar = require('../util/mock-tarball.js')
+
+const { Dir, File } = Tacks
+const testDir = path.join(__dirname, path.basename(__filename, '.js'))
+
+let server
+test('setup mocked registry', t => {
+  common.fakeRegistry.compat({}, (err, s) => {
+    t.ifError(err, 'registry mocked successfully')
+    server = s
+    t.end()
+  })
+})
+
+test('npm repo test-repo-with-directory', t => {
+  const fixture = new Tacks(Dir({
+    'package.json': File({})
+  }))
+  fixture.create(testDir)
+  const packument = {
+    name: 'test-repo-with-directory',
+    'dist-tags': { latest: '1.2.3' },
+    versions: {
+      '1.2.3': {
+        name: 'test-repo-with-directory',
+        version: '1.2.3',
+        dist: {
+          tarball: `${server.registry}/test-repo-with-directory/-/test-repo-with-directory-1.2.3.tgz`
+        },
+        repository: {
+          type: 'git',
+          url: 'git+https://github.com/foo/test-repo-with-directory.git',
+          directory: 'some/directory'
+        }
+      }
+    }
+  }
+  server.get('/test-repo-with-directory').reply(200, packument)
+  return mockTar({
+    'package.json': JSON.stringify({
+      name: 'test-repo-with-directory',
+      version: '1.2.3'
+    })
+  }).then(tarball => {
+    server.get('/test-repo-with-directory/-/test-repo-with-directory-1.2.3.tgz').reply(200, tarball)
+    return common.npm([
+      'repo', 'test-repo-with-directory',
+      '--registry=' + server.registry,
+      '--loglevel=silent',
+      '--browser=' + fakeBrowser
+    ])
+  }).then(([code, stdout, stderr]) => {
+    t.equal(code, 0)
+    t.comment(stdout)
+    t.comment(stderr)
+
+    const res = fs.readFileSync(outFile, 'ascii')
+    t.equal(res, 'https://github.com/foo/test-repo-with-directory/tree/master/some%2Fdirectory\n')
```

Comment:
I expected `.../some/directory`, not `.../some%2Fdirectory`. 🤔
But I find this is a problem of `hosted-git-info` package.
So, I will open a new PR on [npm/hosted-git-info](https://github.com/npm/hosted-git-info). 💪 

The relevant code is here:
https://github.com/npm/hosted-git-info/blob/067fd7f3559fc051fd56528f6231a70ec4e634d0/git-host.js#L38-L40

Reply:
This has been fixed! So this test will actually fail now. We'll fix this up when landing this PR! :D

Reply:
@mikemimik Thanks for your response. I have not touched this PR for a long time, but can I start again?

If you have any advice, I would be glad to accept!

Reply:
It's not necessary to start again. The @npm/cli-team triaged this pull-request today and we're accepting it into `6.14.0`. We can fix up the test when we pull it into the release, it's no worries :)

You noted that `hosted-git-info` does some weird url-encoding, and you had to use "`%2f`". Since then `hosted-git-info` has had some changes and it's fixed. It now returns properly. The change to the test is just changing `%2f` back to `/` :D

Reply:
I understand it. I'm looking forward to a new release 😊